ONSdigital/design-system
# Update husky to v9

### What is the context of this PR?

Fixes: #3297 
Updates husky to v9 and fixes deprecations

### How to review this PR

- Switch to this branch and run `yarn`
- Switch to `main` and see the husky deprecation warning
- Switch back to this branch and see that the warning is gone
- Make a commit locally on this branch (without pushing) and see there is no warning
- Make a commit on another branch and see the warning

This is the warning to look for:
<img width="487" alt="Screenshot 2024-09-12 at 15 30 02" src="https://github.com/user-attachments/assets/f66349c6-4aa3-4dca-8cad-3b4b1b48f073">

### Checklist

This needs to be completed by the person raising the PR.
